### ISSUE TITLE
Support noise models

### ISSUE THREAD

**Issue by speller26** [qiskit-community/qiskit-braket-provider#89]
In Qiskit, it is possible to add a [noise model](https://qiskit.org/documentation/tutorials/simulators/3_building_noise_models.html#Adding-errors-to-a-Noise-Model) to a simulator device so that the noise model is applied to any circuit [run on the device](https://qiskit.org/documentation/tutorials/simulators/3_building_noise_models.html#Executing-the-noisy-simulation). Analogously, in the Amazon Braket SDK, it is possible to build [noise models](https://github.com/aws/amazon-braket-sdk-python/blob/main/src/braket/circuits/noise_model/noise_model.py#L84) that can be applied to circuits to make them into noisy circuits.

### What is the expected behavior?

Amazon Braket DM1 and local density matrix backends should be able to take a Qiskit noise model that will be applied to all circuits run on the device.

### Acceptance criteria

* Adding a `noise_model` kwarg to the Braket provider's `get_backend` method that attaches a noise model to be applied to every circuit run
* Adding `noise_model` and `braket_noise_model` properties to the Braket backends that allow users to inspect the Qiskit and Braket noise models, respectively

```python
dm1 = provider.get_backend("DM1", noise_model=noise_bit_flip)

# Run noisy circuit
circuit = ...  # noiseless circuit
task = dm1.run(circuit, shots=100)  # circuit run with noise model
```

**Comment by rum1887** [comment]
Do other AWS backends support noise inputs from Qiskit?

**Comment by ruijterj** [comment]
Hi , can I work on this issue ?

**Comment by christianbmadsen** [comment]
> Do other AWS backends support noise model input from Qiskit?

No. Today, Braket backends do not support Qiksit noise models. The Braket noise simulators/backends are [`braket_dm`](https://docs.aws.amazon.com/braket/latest/developerguide/braket-devices.html#braket-simulator-dm) and [`DM1`](https://docs.aws.amazon.com/braket/latest/developerguide/braket-devices.html#braket-simulator-dm1).

@rum1887 did you already start work on this issue? If not we will assign to @ruijterj.

**Comment by rum1887** [comment]
> > Do other AWS backends support noise model input from Qiskit?
> 
> No. Today, Braket backends do not support Qiksit noise models. The Braket noise simulators/backends are [`braket_dm`](https://docs.aws.amazon.com/braket/latest/developerguide/braket-devices.html#braket-simulator-dm) and [`DM1`](https://docs.aws.amazon.com/braket/latest/developerguide/braket-devices.html#braket-simulator-dm1).
> 
> @rum1887 did you already start work on this issue? If not we will assign to @ruijterj.

I have not started working yet, I was working out the details of this issue. Can you assign both of us?

**Comment by JordanAWS** [comment]
@rum1887 , @ruijterj , @s-aldaihan 

I've assigned the issue to all of you. Whoever submits a working PR first (including tests) will win the bounty :)

**Comment by JordanAWS** [comment]
Alternatively, you are also welcome to all work together and split the bounty! Just let me know what you prefer.

**Comment by s-aldaihan** [comment]
Thanks@JordanAWS. 
@rum1887 @ruijterj  I am happy to work together if you are interested!

**Comment by JordanAWS** [comment]
Hope things are going well for the team! Does anyone in need support?

**Comment by Newtech66** [comment]
Some questions:

 - `DM1` and `braket_dm` should accept the noise model (if provided) and if other backends are provided it should issue an error saying the backend does not support noise models?
 - If the above backends do not support Qiskit noise models then how should it work? Is there a way to convert Qiskit noise models to Braket noise models and then use that to run the circuit?

**Comment by yitchen-tim** [comment]
The `set_noise_model` method from this PR https://github.com/amazon-braket/amazon-braket-sdk-python/pull/893/files may help with the implementation.
